stages/disks: support resizing an existing partition (#1087)
coreos/ignition: doc/operator-notes.md

| false             | true        | true               | Create specified partition
| true              | false       | false              | Fail
| true              | false       | true               | Delete existing partition

> | true              | true        | false              | Check if existing partition matches the specified one, if it does not match, check if the `resize` field is set to true and all of the specified attributes (`label`, `start`, `uuid`, and `typeGuid`) matches existing partition's attributes except the partition size, then resize an existing partition else fail the execution.

Reviewer: WDYT about this?

```suggestion
| true              | true        | false              | Check if existing partition matches the specified one, if it does not match, check if the `resize` field is set to true and all of the specified attributes (`label`, `start`, `uuid`, and `typeGuid`) matches existing partition's attributes except the partition size, then resize an existing partition otherwise fail.
```

Reviewer 2: Simpler:
```suggestion
| true              | true        | false              | Verify existing partition matches the specified one, otherwise resize it if `resize` is true and partition matches in all respects except size, otherwise fail
```

| true              | true        | false              | Check if existing partition matches the specified one, fail if it does not
| true              | true        | true               | Check if existing partition matches the specified one, delete existing partition and create specified partition if it does not match

### Resize an existing partition

> This is possible only when a partition is present, `shouldExist` is true and `wipePartitionEntry` is false. Ignition will resize an existing partition if the `resize` field is set to true and all of the specified attributes (`label`, `start`, `uuid`, and `typeGuid`) matches the existing partition's attributes except the partition size. 

Reviewer 2: The first sentence suggests that this should go directly into the table entry.
